packages/ckeditor5-restricted-editing/src/restrictededitingmode/converters.ts

```diff
@@ -108,7 +108,7 @@ export function extendMarkerOnTypingPostFixer( editor: Editor ): ModelPostFixer
 		let changeApplied = false;
 
 		for ( const change of editor.model.document.differ.getChanges() ) {
-			if ( change.type == 'insert' && change.name == '$text' ) {
+			if ( change.type == 'insert' && ( change.name == '$text' || change.name == 'imageInline' ) ) {
```

Comment:
This would not work for other inline widgets. Please use schema to check if it is an inline object allowed where text.

Reply:
Is it possible to verify element in `DiffItemInsert` with schema?
![image](https://github.com/ckeditor/ckeditor5/assets/47000454/741b1510-201f-46eb-abef-00771d877dcc)

Reply:
Nevermind, indeed there is a way, `schema.isInline( change.position.nodeAfter! )`